models/base_model.go

```diff
@@ -0,0 +1,77 @@
+package models
+
+import (
+	"errors"
+
+	"github.com/sirupsen/logrus"
+	"github.com/transcovo/go-chpr-logger"
+	mgo "gopkg.in/mgo.v2"
+)
+
+/*
+ErrGenericDB is the error returned when an unknown mongo error happens.
+*/
+var ErrGenericDB = errors.New("DB Error")
```

Comment:
I disagree with this: you will never nee to test that someting is a "generic db error", you will rather test some specific errors and have a catchall that handle "all the other errors". Any usually you will log the error and go on. If you hide the actual error behind the ErrGenericDB, it will make the log much less useful. Also, event if you log from this lib before returning the generic error, the log will be less useful because it will not contain the context which caused this error.
So I think the best thing to do is to return the original error.